tests/SchemaParserTest.php

```diff
@@ -73,24 +87,29 @@ class TestEntity
     public array $hashmap = [];
 
     public TestSubEntity $class;
+
+    /**
+     * @var int|string|null
+     *
+     * @OA\Union(types={ @OA\IntegerValue(minimum=2), @OA\StringValue(minLength=10) })
+     */
+    public $scalarUnion;
+
+    /**
+     * @var AEntity|BEntity
+     *
+     * @OA\Union(discriminator="type", mapping={ "a":"AEnt", "b":"BEnt" })
```

Comment:
uprimne tady moc nerozumim tomu, proc tady musime pouzivat ty schemaNames? - potazmo ty jsou pro me celkove zahadou - proc nenapisu `mapping={ "a": AEntity::class, "b": BEntity::class }`?

obecne jsem to bral tak ze tohle rozhrani by o tom, ze existujuji nejaka schemata komponent nebudou vedet - navic mi teda tady troch chybi ten pristup te "magie" o ktere jsme se bavili, ze na entite mam definovany tu value toho diskriminatoru a ten Union si to nacucne sam, pokud teda rucne nema ten mapping definovany

Reply:
jak to pak priradis? ObjectSchema neni nijak svazane s class